frontends/concrete-python/concrete/fhe/mlir/context.py

```diff
@@ -1015,6 +1015,88 @@ def convert_to_chunks_and_map(
 
         return result
 
+    def pack_multivariate_inputs(self, xs: List[Conversion]) -> Conversion:
+        """
+        Packs inputs of multivariate table lookups.
+
+        Args:
+            xs (List[Conversion]):
+                operands
+
+        Returns:
+            Conversion:
+                packed operands
+        """
+
+        assert all(x.is_encrypted for x in xs)
+
+        required_bit_width = sum(x.original_bit_width for x in xs)
+        maximum_bit_width = max(x.bit_width for x in xs)
+        bit_width_to_pack = max(required_bit_width, maximum_bit_width)
+
+        shifted_xs = []
+        next_shift_amount = 0
+
+        for x in xs:
+            needs_cast = x.bit_width != bit_width_to_pack
+            needs_sanitize = x.is_signed
+            needs_shift = next_shift_amount > 0
+
+            shift_amount = next_shift_amount
+            next_shift_amount += x.original_bit_width
+
+            if not any([needs_cast, needs_sanitize, needs_shift]):
```

Comment:
`if not(needs_cast and need... and ...):` is easier to me

Reply:
Sure.